sys/dev/nvme/nvme_ctrlr_cmd.c

```diff
@@ -251,6 +253,65 @@ nvme_ctrlr_cmd_set_interrupt_coalescing(struct nvme_controller *ctrlr,
 	    0, 0, 0, 0, NULL, 0, cb_fn, cb_arg);
 }
 
+void
+nvme_ctrlr_cmd_set_apst(struct nvme_controller *ctrlr, uint32_t itpt_factor,
+    uint32_t max_latency, nvme_cb_fn_t cb_fn, void *cb_arg)
+{
+	struct nvme_completion_poll_status status;
+	struct nvme_power_state *ps;
+	uint64_t *entry, total_latency;
+	uint32_t cdw11, itps, itpt;
+
+	if (ctrlr->cdata.apsta == 0 || ctrlr->quirks & QUIRK_DISABLE_APST) {
+		nvme_printf(ctrlr, "APST is not supported by the device\n");
+		return;
+	}
+
+	if (ctrlr->cdata.npss > 31) {
+		nvme_printf(ctrlr, "invalid NPSS, APST is not set\n");
```

Comment:
Boot verbose, I think. Or maybe this should be a kassert if it's a programming error that originates in the kernel.

Reply:
Thanks, I'll change it to `KASSERT`.